src/engine.cc

```diff
@@ -132,9 +132,11 @@ void EngineController::UpdateFromUciOptions() {
     if (!syzygy_tb_->init(tb_paths)) {
       CERR << "Failed to load Syzygy tablebases!";
       syzygy_tb_ = nullptr;
-    } else {
-      tb_paths_ = tb_paths;
     }
+    tb_paths_ = tb_paths;
+  } else if (tb_paths.empty() && syzygy_tb_) {
+    syzygy_tb_ = nullptr;
+    tb_paths_.clear();
```

Comment:
I think there is still a case - a failed syzygy load will mean syzygy_tb_ is null but tb_paths_ is not already empty.  And we end up with a stale tb_paths_ value...

Reply:
Since `tb_paths_` is only used to check whether the UCI option changed there is no problem, but if we clear it the next `go` command will try lo load the tb again.

Reply:
this sequence.
set tb_path  to invalid directory.
set tb_path to empty
set tb_path to same invalid directory (which is now fixed) - the first if branch is skipped because tb_paths_ still contains the old invalid directory value.  Thus tb doesn't load.

Reply:
So remove `&& syzygy_tb_` from the `else if`? Seems reasonable.

Reply:
Sounds good.